.github/workflows/build-push.yml

```diff
@@ -0,0 +1,97 @@
+# This action script will take a merge to main and build
+# the neuro-san-demos docker image and push it to our ECR.
+# It will then update the manifest file with the new tag
+# which then will get picked up by our Argo CD system and 
+# deploy it to the dev environment.
+# In the case of a github Release event, the similar flow
+# occurs, but in this case, the staging environment is updated.
+# This script does not automatically push anything to production;
+# that case is handled manually.
+# Note, this script only updates the neuro-san-demos image, while
+# the deploy also uses a custom ui image; that process is still 
+# manual as the custom ui build is a manual process and not 
+# run frequently.
+
+name: Build and Push Docker Image to ECR and promote to manifest repo
+
+on:
+  push:
+    branches:
+      - main
+  release:
+    types: [published]  # Triggers when a GitHub Release is published
+  workflow_dispatch:  # Allow manual runs if needed
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    outputs:
+      image_tag: ${{ steps.set-vars.outputs.image_tag }}
+      target_environment: ${{ steps.set-vars.outputs.target_environment }}
+
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        
+      - name: Set up variables
+        id: set-vars
+        run: |
+          if [ "${{ github.event_name }}" = "release" ]; then
+            echo "image_tag=${{ github.event.release.tag_name }}" >> $GITHUB_OUTPUT
```

Comment:
So we get the image tag from the GitHub release now?
Does it mean we need to update `deploy/build.sh`? Because it's currently doing:
```
export SERVICE_VERSION=${SERVICE_VERSION:-0.0.9}
```